tests/unit-tests/internal/emails/generators/test-class-student-submits-quiz.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace SenseiTest\Internal\Emails;
+
+use Sensei\Internal\Emails\Email_Repository;
+use Sensei\Internal\Emails\Student_Submits_Quiz;
+use Sensei_Factory;
+
+/**
+ * Tests for Sensei\Internal\Emails\Student_Submits_Quiz class.
+ *
+ * @covers \Sensei\Internal\Emails\Student_Submits_Quiz
+ */
+class Student_Submits_Quiz_Test extends \WP_UnitTestCase {
+	use \Sensei_Course_Enrolment_Test_Helpers;
+	use \Sensei_Course_Enrolment_Manual_Test_Helpers;
+
+	/**
+	 * Factory for creating test data.
+	 *
+	 * @var Sensei_Factory
+	 */
+	protected $factory;
+
+	/**
+	 * Email repository instance.
+	 *
+	 * @var Email_Repository
+	 */
+	protected $email_repository;
+
+	public function setUp(): void {
+		parent::setUp();
+		$this->prepareEnrolmentManager();
+
+		$this->factory          = new Sensei_Factory();
+		$this->email_repository = new Email_Repository();
+	}
+
+	/**
+	 * Clean up after all tests.
+	 */
+	public static function tearDownAfterClass(): void {
+		parent::tearDownAfterClass();
+		self::resetEnrolmentProviders();
+	}
+
+	public function testGenerateEmail_WhenCalledByStudentSubmittedQuizEvent_CallsEmailSendingActionWithRightData() {
```

Comment:
```suggestion
	public function testStudentSubmitsQuizMailToTeacher_WhenCalled_CallsEmailSendingActionWithRightData() {
```
I think the beginning of the name was just the copy-paste issue and needs to be fixed.
As for the rest. This test is an integration one. Honestly, I don't see any reasons for that: we can directly call the `student_submits_quiz_mail_to_teacher` method and check results without additional hacks.